packages/x-date-pickers/src/internals/hooks/useField/useFieldState.ts

```diff
@@ -200,10 +200,15 @@ export const useFieldState = <
 
   const clearActiveSection = () => {
     if (selectedSectionIndexes == null) {
-      return undefined;
+      return;
     }
 
     const activeSection = state.sections[selectedSectionIndexes.startIndex];
+
+    if (activeSection.value === '') {
```

Comment:
We currently fire `onChange` even when pressing `Backspace` on an already empty section, which is not useful